### PR TITLE
Upgrade Node backend to Beta Client Libraries

### PR DESCRIPTION
**Pull request by lindawoo-plaid**
Updates Node backend from 7.x.x API Client Libraries to the 9.0.0-beta.11 Beta version.

### CODE REVIEW

File: node/index.js [hunk 1]
```diff
@@ -437,64 +255,8 @@ const prettyPrintResponse = (response) => {
...
-            pdf: assetReportGetPdfResponse.buffer.toString('base64'),
-          });
-        },
-      );
-    },
-  );
+const makeErrorObject = (error) => {
+  return {
+    error: { ...error.data, status_code: error.status },
+  };
```
Comment (skylarmb) [inline node/index.js:261]: we have helper functions in at least ruby and python that do this same thing and they are called `formatError`. We could use the same name for this function for consistency!

File: node/package.json [hunk 2]
```diff
@@ -18,6 +18,6 @@
     "express": "4.16.x",
     "moment": "2.22.x",
     "nodemon": "^2.0.5",
-    "plaid": "6.x.x"
+    "plaid": "^9.0.0-beta.11"
```
Comment (skylarmb) [inline node/package.json:21]: 👏

**Review by skylarmb — APPROVED** [review]
nice! overall looks good! 

1) should the `create_link_token_for_payment` and `assets` endpoints just be modernized instead of deleted? they are going to be used shortly right?

2) could you just get someone from the OpenAPI crew (chase, stephen) to give you a 👍 on how you are constructing the `new PlaidApi` / `new Configuration` etc stuff? I am just not sure what the blessed way of integrating with this new generated lib is, so would love a quick 👀 from them!

Nice work!

File: node/index.js [hunk 3]
```diff
@@ -96,337 +92,159 @@ app.post('/api/create_link_token', function (request, response, next) {
...
-                }
-                response.json(createTokenResponse);
-              },
-            );
-          },
-        );
-      },
-    );
-  },
-);
```
[the remark this answers is not in the document]
Comment (lindawoo-plaid) [reply node/index.js:185]: okay, i've modernized them and put them back in

File: node/index.js (hunk 1 above)
Comment (cgfarmer4) [reply node/index.js:461]: We rm'd these helper functions. They are not generated.

File: node/index.js [hunk 4]
```diff
@@ -1,11 +1,10 @@
 // read env vars from .env file
 require('dotenv').config();
-
+const { Configuration, PlaidApi, PlaidEnvironments } = require('plaid');
```
Comment (cgfarmer4) [inline node/index.js:3]: we could mod the `package.json` to make these imports but this is fine too.

File: node/index.js [hunk 5]
```diff
@@ -50,15 +49,20 @@ let PAYMENT_ID = null;
...
 // Initialize the Plaid client
 // Find your API keys in the Dashboard (https://dashboard.plaid.com/account/keys)
-const client = new plaid.Client({
-  clientID: PLAID_CLIENT_ID,
-  secret: PLAID_SECRET,
-  env: plaid.environments[PLAID_ENV],
-  options: {
-    version: '2019-05-29',
+
+const configuration = new Configuration({
```
Comment (stephenjayakar) [inline node/index.js:53]: 🚀

File: node/index.js [hunk 6]
```diff
@@ -96,250 +100,252 @@ app.post('/api/create_link_token', function (request, response, next) {
...
+        },
+        client_name: 'Plaid Quickstart',
+        products: PLAID_PRODUCTS,
+        country_codes: PLAID_COUNTRY_CODES,
+        language: 'en',
+        payment_initiation: {
+          payment_id: PAYMENT_ID,
+        },
+      };
+      if (PLAID_REDIRECT_URI !== '') {
```
Comment (stephenjayakar) [inline node/index.js:159]: completely optional and for your consideration as this isn't necessary a better pattern (more concise though):

```node
configs = {
...
  redirect_uri = PLAID_REDIRECT_URI ? PLAID_REDIRECT_URI : undefined
...
}
```

benefits: 
* you can make `configs` immutable
* no conditional inserts after `configs`

drawbacks: 
* ternaries detract from clarity imo
* may or may not cause issues storing `undefined` directly into an `object` :'). we will see.

Comment (stephenjayakar) [reply node/index.js:159]: this pattern also shows up throughout the code, so whatever your decision propagate it please!

File: node/index.js [hunk 7]
```diff
@@ -96,250 +100,252 @@ app.post('/api/create_link_token', function (request, response, next) {
...
-        city: 'Little Whinging',
-        postal_code: '11111',
-        country: 'GB',
-      },
-      function (error, createRecipientResponse) {
-        const recipientId = createRecipientResponse.recipient_id;
+  async function (request, response, next) {
+    try {
+      const createRecipientResponse = await client.paymentInitiationRecipientCreate(
+        {
```
Comment (stephenjayakar) [inline node/index.js:121]: there's a mix of inline request objects v.s. defining the objects on their own line (`config = {...` pattern). whichever pattern you prefer, should stick with it! 

i personally prefer inline as it's more concise, but the separate line has a clear benefit for being more debuggable.

**Review by stephenjayakar — COMMENTED** [review]
great work @lindawoo-plaid, thanks for adopting our beta :). made some suggestions, feel free to DM me for more context. also, DM topic but is this going to master or is this going to be like quickstart-beta?

File: node/index.js (hunk 7 above)
Comment (lindawoo-plaid) [reply node/index.js:121]: done. thanks!

File: node/index.js [hunk 8]
```diff
@@ -96,337 +92,159 @@ app.post('/api/create_link_token', function (request, response, next) {
...
-    response.json(createTokenResponse);
-  });
+  try {
+    const createTokenResponse = await client.linkTokenCreate(configs);
+    response.json(createTokenResponse.data);
+  } catch (error) {
+    prettyPrintResponse(error);
+    return response.json({
+      error: error.response.data,
+    });
```
[the remark this answers is not in the document]
Comment (lindawoo-plaid) [reply node/index.js]: yes...

File: node/index.js (hunk 1 above)
Comment (lindawoo-plaid) [reply node/index.js:461]: i renamed.

File: node/index.js (hunk 6 above)
Comment (lindawoo-plaid) [reply node/index.js:158]: I think I am going to leave it cause I don't want any issues with storing an undefined property...

**Review by skylarmb — APPROVED** [review]

**Pull request merged**